trie/database.go

```diff
@@ -781,17 +781,31 @@ func (db *Database) Update(nodes *MergedNodeSet) error {
 	defer db.lock.Unlock()
 	// Insert dirty nodes into the database. In the same tree, it must be
 	// ensured that children are inserted first, then parent so that children
-	// can be linked with their parent correctly. The order of writing between
-	// different tries(account trie, storage tries) is not required.
-	for owner, subset := range nodes.sets {
-		for _, path := range subset.updates.order {
-			n, ok := subset.updates.nodes[path]
-			if !ok {
-				return fmt.Errorf("missing node %x %v", owner, path)
+	// can be linked with their parent correctly.
+	//
+	// Note, the storage tries must be flushed before the account trie to
+	// retain the invariant that children go into the dirty cache first.
+	var order []common.Hash
+	for owner := range nodes.sets {
+		if owner == (common.Hash{}) {
+			continue
+		}
+		order = append(order, owner)
+	}
+	if _, ok := nodes.sets[common.Hash{}]; ok {
+		order = append(order, common.Hash{})
```

Comment:
Here, the trie with owner empty representing the account trie is pushed to the back of the array so that all the storage dirty nodes are flushed first.
Reference: https://github.com/ethereum/go-ethereum/commit/5758d1fb1162f14ebb2420017d83181e576e3d88

Reply:
Add this like comment in the source code too ?

Reply:
There is a comment like this above
```
 	// Note, the storage tries must be flushed before the account trie to
	// retain the invariant that children go into the dirty cache first.
```
Do you think this is clear enough?

Reply:
```the trie with owner empty representing the account trie is pushed to the back of the array so that all the storage dirty nodes are flushed first.``` I think u can add those blocks for showing why we use common.Hash{} here.